frontend/app/src/next/components/runs/run-event-log/run-event-log.tsx

```diff
@@ -0,0 +1,509 @@
+import {
+  V1WorkflowRun,
+  V1TaskEvent,
+  V1TaskEventType,
+  V1TaskSummary,
+} from '@/lib/api';
+import { useRunDetail } from '@/next/hooks/use-run-detail';
+import { cn } from '@/next/lib/utils';
+import { RunStatusConfigs } from '../runs-badge';
+import { WorkflowRunStatus } from '@/lib/api';
+import {
+  CheckCircle2,
+  PlayCircle,
+  Clock,
+  AlertCircle,
+  XCircle,
+  RefreshCw,
+  PauseCircle,
+  Timer,
+  UserCog,
+  Unlock,
+  RotateCw,
+  Send,
+  AlertTriangle,
+  Bell,
+  Plus,
+  SkipForward,
+  ArrowUpRight,
+  Info,
+  Cpu,
+  Bug,
+  CpuIcon,
+} from 'lucide-react';
+import { VscJson } from 'react-icons/vsc';
+import { Button } from '@/next/components/ui/button';
+import { Time } from '@/next/components/ui/time';
+import { useMemo } from 'react';
+import { RunId } from '../run-id';
+import {
+  FilterGroup,
+  FilterText,
+  FilterSelect,
+} from '@/next/components/ui/filters/filters';
+import { useFilters } from '@/next/hooks/utils/use-filters';
+import { ROUTES } from '@/next/lib/routes';
+
+interface RunEventLogProps {
+  workflow: V1WorkflowRun;
+  onTaskSelect?: (
+    taskId: string,
+    options?: Parameters<typeof ROUTES.runs.taskDetail>[2],
+  ) => void;
+}
+
+type EventConfig = {
+  icon: React.ComponentType<{ className?: string }>;
+  message: string | ((event: V1TaskEvent) => string);
+  showWorkerButton?: boolean;
+  status: WorkflowRunStatus;
+  title: string;
+};
+
+const LogEventType = 'LOG_LINE' as V1TaskEventType;
+
+interface ActivityFilters {
+  search?: string;
+  eventType?: V1TaskEventType[];
+  taskId?: string[];
+}
+
+const EVENT_CONFIG: Record<V1TaskEventType, EventConfig> = {
```

Comment:
would be great to make this a function with a `switch` case with an exhaustiveness check too I think

Reply:
i prefer Record over switch cases for these sorts of things